entry.go

```diff
@@ -122,46 +128,8 @@ func (te *TunaEntry) Start() {
 			}
 		}()
 
-		go func() {
-			var np *nkn.NanoPay
-			for {
-				time.Sleep(DefaultNanoPayUpdateInterval)
-				bytesIn := atomic.LoadUint64(&te.bytesIn)
-				bytesOut := atomic.LoadUint64(&te.bytesOut)
-				entryToExitPrice, exitToEntryPrice := te.GetPrice()
-				delta := exitToEntryPrice*common.Fixed64(bytesIn-te.bytesInPaid)/TrafficUnit + entryToExitPrice*common.Fixed64(bytesOut-te.bytesOutPaid)/TrafficUnit
-				if delta == 0 {
-					continue
-				}
-				paymentReceiver := te.GetPaymentReceiver()
-				if np == nil || np.Address() != paymentReceiver {
-					var err error
-					np, err = te.Wallet.NewNanoPay(paymentReceiver, te.config.NanoPayFee, DefaultNanoPayDuration)
-					if err != nil {
-						continue
-					}
-				}
-				tx, err := np.IncrementAmount(delta.String())
-				if err != nil {
-					continue
-				}
-				txData := tx.ToArray()
-				session, err := te.getSession(false)
-				if err != nil {
-					continue
-				}
-				stream, err := session.OpenStream()
-				if err != nil {
-					continue
-				}
-				n, err := stream.Write(txData)
-				if n == len(txData) && err == nil {
-					te.bytesInPaid = bytesIn
-					te.bytesOutPaid = bytesOut
-				}
-				stream.Close()
-			}
-		}()
+		session, err := te.getSession(false)
```

Comment:
It's better to call `getSession` when use it (like the code you deleted) because previous session might be closed. Also it's better to change `sendNanoPayment()` to `te. sendNanoPayment()` so you don't need to pass so many arguments here

Reply:
So we need to add `getSession` function to the `TunaExit` also?

Reply:
There is already one, see https://github.com/nknorg/tuna/pull/6/files#diff-32f22e4789509e4288e8ee93f532fbd0L149

Reply:
Nevermind, just realized you are trying to combine duplicate code

Reply:
You can use something like in entry:
```
go func() {
for {
    session, err := te.getSession(false)
    sendNanoPayment(session, ...)
}
}()
```
basically extract the for loop outside the function

Reply:
And you forget to handle error here